### PR TITLE
Add support for `ansi` colors and add `.ansi256ToAnsi()`, `.rgbToAnsi()` and `.hexToAnsi()`

### PR DESCRIPTION
**Pull request by Richienb**
https://github.com/chalk/chalk/pull/433#issuecomment-822089880

### CODE REVIEW

**Review by sindresorhus — APPROVED** [review]

**Comment by sindresorhus** [comment]
@Qix- LGTY?

File: readme.md [hunk 1]
```diff
@@ -112,21 +113,25 @@ console.log(styles.codes.get(36));
 //=> 39
 ```
 
-## [256 / 16 million (TrueColor) support](https://gist.github.com/XVilka/8346728)
+## [16 / 256 / 16 million (TrueColor) support](https://gist.github.com/XVilka/8346728)
```
Comment (Qix-) [inline readme.md:116]:
```suggestion
## 16 / 256 / 16 million (TrueColor) support
```

File: readme.md [hunk 2]
```diff
@@ -112,21 +113,25 @@ console.log(styles.codes.get(36));
...
 
-`ansi-styles` allows converting between various color formats and ANSI escapes, with support for 256 and 16 million colors.
+`ansi-styles` allows converting between various color formats and ANSI escapes, with support for 16, 256 and 16 million colors.
```
Comment (Qix-) [inline readme.md:118]:
```suggestion
`ansi-styles` allows converting between various color formats and ANSI escapes, with support for 16, 256 and [16 million colors](https://gist.github.com/XVilka/8346728).
```

File: index.js [hunk 3]
```diff
@@ -149,6 +153,58 @@ function assembleStyles() {
...
+					red = (((code - 232) * 10) + 8) / 255;
+					green = red;
+					blue = red;
+				} else {
+					code -= 16;
+
+					const remainder = code % 36;
+
+					red = Math.floor(code / 36) / 5;
+					green = Math.floor((code % 36) / 6) / 5;
```
Comment (Qix-) [inline index.js:181]:
```suggestion
					green = Math.floor(remainder / 6) / 5;
```

**Review by Qix- — CHANGES_REQUESTED** [review]
A few nits

**Comment by Richienb** [comment]
The CI is yet to have cached a version of `@babel/traverse` released 44 minutes ago so it is currently failing as of now.

**Comment by Qix-** [comment]
Can we get some tests for `ansi256To16` specifically?

**Comment by Richienb** [comment]
> Can we get some tests for `ansi256ToAnsi` specifically?
> 
> Otherwise, LGTM. Thanks for the heads up about the failing test.

Since this function is being added to `color-convert` (https://github.com/Qix-/color-convert/issues/92) it should really be tested there. That's why there aren't any tests here for rgb → ansi256.

**Comment by Qix-** [comment]
I disagree, but okay. Chalk does not rely on color convert any longer so their testing concerns no longer overlap. If you're going to vendor code, you should include automated tests in the destination codebase, IMO.

**Comment by Richienb** [comment]
If it's a direct copy-paste from the original codebase that is already tested, then IMO unit tests aren't nessessary. Now would be the time to add the short-circut to `color-convert`.

**Comment by sindresorhus** [comment]
Sorry, I missed the comment about tests. I agree we should have some tests here too.

**Pull request merged**